diff --git a/src/evict/evict_lru.c b/src/evict/evict_lru.c
index a04bbc2a2a..b88b49bd49 100644
--- a/src/evict/evict_lru.c
+++ b/src/evict/evict_lru.c
@@ -1563,17 +1563,24 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
         else if (candidates > entries / 2)
             queue->evict_candidates = candidates;
         else {
-            /*
-             * Take all of the urgent pages plus a third of ordinary candidates (which could be
-             * expressed as WTI_EVICT_WALK_INCR / WTI_EVICT_WALK_BASE). In the steady state, we want
-             * to get as many candidates as the eviction walk adds to the queue.
-             *
-             * That said, if there is only one entry, which is normal when populating an empty file,
-             * don't exclude it.
-             */
-            queue->evict_candidates = 1 + candidates + ((entries - candidates) - 1) / 3;
-            if (queue->evict_candidates > entries / 2)
-                queue->evict_candidates = entries / 2;
+            if (F_ISSET(evict, WT_EVICT_CACHE_UPDATES) &&
+              evict->evict_empty_score > WT_EVICT_SCORE_CUTOFF) {
+                /* Take all available candidates when in update mode with consistently empty queues
+                 */
+                queue->evict_candidates = entries;
+            } else {
+                /*
+                 * Take all of the urgent pages plus a third of ordinary candidates (which could be
+                 * expressed as WTI_EVICT_WALK_INCR / WTI_EVICT_WALK_BASE). In the steady state, we
+                 * want to get as many candidates as the eviction walk adds to the queue.
+                 *
+                 * That said, if there is only one entry, which is normal when populating an empty
+                 * file, don't exclude it.
+                 */
+                queue->evict_candidates = 1 + candidates + ((entries - candidates) - 1) / 3;
+                if (queue->evict_candidates > entries / 2)
+                    queue->evict_candidates = entries / 2;
+            }
 
             evict->read_gen_oldest = read_gen_oldest;
         }
